inc/shortcodes/glossary/class-glossary.php

```diff
@@ -0,0 +1,268 @@
+<?php
+/**
+ * @author   Brad Payne, Alex Paredes
+ * @license  GPLv3 (or any later version)
+ */
+
+namespace Pressbooks\Shortcodes\Glossary;
+
+use PressbooksMix\Assets;
+
+class Glossary {
+
+	/**
+	 * @var Glossary
+	 */
+	static $instance = null;
+
+	/**
+	 * @var array
+	 */
+	static $glossary_terms = [];
+
+	/**
+	 * Function to init our class, set filters & hooks, set a singleton instance
+	 *
+	 * @return Glossary
+	 */
+	static public function init() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+			self::hooks( self::$instance );
+			self::setGlossaryTerms();
```

Comment:
I don't think `setGlossaryTerms` should in init(),

If a book has 10,000 glossary items this implies that 10,000 posts will be loaded on every page refresh. Even when not needed. This is going to cause performance issues.

Can you explore changing:

 + `self::setGlossaryTerms()` to `self::getGlossaryTerms()`
 + `$foo = self::$glossary_terms` to $foo =self::getGlossaryTerms()`
 + Caching when invoking getGlossaryTerms ( [`// cheap cache`  is probably good enough here?](https://github.com/pressbooks/pressbooks/blob/15f946b1a976bf0082d50bbcc33047f2a9be0679/compatibility.php#L21) I would like to see what happens with a glossary of 10,000 items but I'm sure we'll see it in Production and New Relic sooner than later )

I also don't think these needs to be static. All calls to  self::$glossary_terms that I see can be done in `$this` scope? If it doesn't work in `'glossary_terms'     => wp_json_encode( self::$glossary_terms )` because of the closure then `Glossary::init()->getGlossaryTerms()` would work fine.